agent/doc/UML/hydra_lab_exception_monitor_design.puml

```diff
@@ -0,0 +1,62 @@
+@startuml exception_monitor
+interface ExceptionReporter {
+    + void reportException(Exception e)
+    + void reportException(Exception e ,Thread thread)
+    + void reportException(String message, StackTraceElement[] stackTrace, String threadName)
```

Comment:
Could you share the reason why we still need the following method?
reportException(String message, StackTraceElement[] stackTrace, String threadName)

Reply:
Yes, we can't get the Exception and Thread object. 
![image](https://github.com/microsoft/HydraLab/assets/26757995/698ab089-b76b-4c5c-992b-7f844f1902de)

Reply:
Could you try to construct an Exception with the data? Following 伪代码：
```java
// You may create a new Exception type "LoggerCaptureException"
exception = new Exception("Exception collected in Logger "+ getLoggerName()+" on thread "+ getthreadname)
// you may leverage the Exception message to carry more data like level, marker, message formattedMessage, etc.
// THis can be done by adding fields in LoggerCaptureException, that way we can carry all the data in the structure. Do n't have to append it all in the message.
exception.setStacktrace(getCallerdata())
```